stackit/stackit_core.py

```diff
@@ -39,24 +39,31 @@ def focusQuestion(questions, count):
     userInput = '0'
     #Looping while the user wants to see more input
     while(userInput != 'm'):
-        userInput = promptUser("Press m for more, or a number to select: ")
-        if(userInput == 'm'):
-            break
-        if(0 < int(userInput) and int(userInput) <= count):
-            print("\n\n\n\n\n\n")
-            printFullQuestion(questions[int(userInput)- 1])
-            branchInput = '0'   #user deciding whether to branch to open browser, or to return to search
-            while(branchInput != 'x'):
-                branchInput = promptUser("Press b to launch browser, or x to return to search: ")
-                if(branchInput == 'b'):
-                    webbrowser.open(questions[int(userInput)-1].json['link'], new=0, autoraise=True)
-            #User selects x to return to search
-            if(branchInput == 'x'):
-                print("\n\n\n\n\n\n\n\n\n\n\n\n")
-                #Ranging over the 5 questions including the user's choice
-                for j in range(5*int((int(userInput)-1)/5), 5*int((int(userInput)-1)/5)+5):
-                    printQuestion(questions[j], j+1)
-                continue   #exit the inner while loop
+        userInput = promptUser("Enter m for more, a question number to select, or q to exit: ")
+        try:
+            if(userInput == 'q'):
+                sys.exit()
+            if(userInput == 'm'):
+                break
+            if(0 < int(userInput) and int(userInput) <= count):
+                print("\n\n\n\n\n\n")
+                printFullQuestion(questions[int(userInput)- 1])
+                branchInput = '0'   #user deciding whether to branch to open browser, or to return to search
+                while(branchInput != 'x'):
+                    branchInput = promptUser("Enter b to launch browser, or x to return to search: ")
+                    if(branchInput == 'b'):
+                        webbrowser.open(questions[int(userInput)-1].json['link'], new=0, autoraise=True)
+                #User selects x to return to search
+                if(branchInput == 'x'):
+                    print("\n\n\n\n\n\n\n\n\n\n\n\n")
+                    #Ranging over the 5 questions including the user's choice
+                    for j in range(5*int((int(userInput)-1)/5), 5*int((int(userInput)-1)/5)+5):
+                        printQuestion(questions[j], j+1)
+                    continue   #exit the inner while loop
+        except:
+            if (userInput != 'q'):
+                print("The input entered was not recognized as a valid choice.  Exiting...")
+            sys.exit()
```

Comment:
Would rather this point returned to the initial prompt––"Enter m for more, a question number to select, or q to exit: "––instead of sys.exit()-ing. Allows for user mistakes. There is a manual sys.exit() option in the input 'q' now, as well as control-C.

tl;dr: would rather not quit without explicit user request.